### PR TITLE
Build test darwin target

### LINKED ISSUE #19569: Catch ginkgo test regressions on macOS
We've had [instances](https://github.com/cilium/cilium/commit/3222f50d05d722e39563fa65fe99b52c0858941a) of [PRs](https://github.com/cilium/cilium/commit/ea75a7204849356d5bb2cfe3230b44d1ef2c32c2) breaking ginkgo test runs on macOS. Add a GitHub action for `darwin` target to catch such regressions on the PRs introducing platform specific code changes. 

Some of the options that were discussed for the GH action - 

- Run `GOOS=darwin go build` in the `test` directory as ginkgo does.
- `GOOS=darwin ginkgo -dryRun`

### PR DESCRIPTION
**Pull request by aditighag**
This PR subsumes https://github.com/cilium/cilium/pull/21568. 

(Copied original reviewers.)

Fixes: https://github.com/cilium/cilium/issues/19569

### CODE REVIEW

**Comment by aditighag** [comment]
Maybe we can skip the removal of test artifacts from the build target. The `clean` target should do that.

Screen Shot 2023-01-25 at 2 51 25 PM

**Review by christarazi — APPROVED** [review]

File: .github/workflows/lint-build-commits.yaml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
             ${{ github.event.pull_request.commits_url }})
           PR_FIRST_SHA=$(echo "$PR_COMMITS_API_JSON" | jq -r ".[0].sha")
           PR_PARENT_SHA=$(git rev-parse "${PR_FIRST_SHA}^")
-          git rebase --exec "make -C test build -j $(nproc)" $PR_PARENT_SHA
+          git rebase --exec "make -C test build -j $(nproc);make -C test build-darwin" $PR_PARENT_SHA
```
Comment (joestringer) [inline .github/workflows/lint-build-commits.yaml]: I think that if the first build fails, then we should cancel the second?

```suggestion
          git rebase --exec "make -C test build -j $(nproc) && make -C test build-darwin" $PR_PARENT_SHA
```

**Comment by aditighag** [comment]
> @aditighag as long as the test file being in the repo doesn't prevent the darwin build from running - that's why I assumed that the rm was happening in between before.

Good point! The `rm` step was only added to the darwin target. Anyway I don't think it's a problem. 

Both the targets are compiled - https://github.com/cilium/cilium/actions/runs/4010655158/jobs/6887402660

File: test/Makefile [hunk 2]
```diff
@@ -29,6 +29,10 @@ build:
 	$(GINKGO) build -tags=integration_tests
 	$(QUIET)$(MAKE) -C bpf/
 
+build-darwin:
+	@$(ECHO_GINKGO)$@
+	GOOS=darwin $(GINKGO) build -tags=integration_tests
```
Comment (joestringer) [inline test/Makefile:34]: Woops, missed this one before. `GOOS=darwin $(GINGKO) ...` can end up being translated into `GOOS=darwin @ ginkgo ...`, which I think prevents Make from interpreting the `@` (for quiet). Ideally whichever binaries are defined by make variables, those should appear at the start of the line. Maybe we can move the `GOOS` env var to the end of the line?

```suggestion
	$(GINKGO) build -tags=integration_tests GOOS=darwin
```

**Review by nbusseneau — APPROVED** [review]

File: test/Makefile (hunk 2 above)
Comment (aditighag) [reply test/Makefile:34]: >Maybe we can move the GOOS env var to the end of the line?

That doesn't work. 

```
ginkgo build -tags=integration_tests GOOS=darwin
Found no test suites
For usage instructions:
        ginkgo help
make: *** [build-darwin] Error 1
```

Comment (joestringer) [reply test/Makefile:34]: Ah okay. In that case, I think I'm the main one affected by the quiet output so maybe we just merge this as-is and I can play around with it later if it's bothering me.

**Pull request merged**